### ISSUE TITLE
Setting parent map after child map has no effect.

### ISSUE THREAD

**Issue by Ambrevar** [emacs-evil/evil#938]
# Issue type

- Bug report

# Environment

Emacs version: GNU Emacs 25.2.2 (x86_64-unknown-linux-gnu, GTK+ Version 3.10.8) of 2017-06-28
Operating System: Ubuntu Trusty
Evil version: 1.2.13
Evil installation type: MELPA
Graphical/Terminal: Graphical
Tested in a `make emacs` session (see [CONTRIBUTING.md](https://github.com/emacs-evil/evil/blob/master/CONTRIBUTING.md#bug-reports)): No

# Reproduction steps
- Start Emacs
- Install pdf-tools
- Run

```
(evil-define-key 'motion pdf-view-mode-map
  "j" 'pdf-view-next-line-or-next-page)

(evil-define-key 'motion image-mode-map
  "j" 'image-next-line)
```

- Open an image file.

# Expected behavior

`C-h c j` returns `image-next-line`.

# Actual behavior

`C-h c j` returns `pdf-view-next-line-or-next-page`.

# Further notes

`pdf-view-mode-map` inherits from `image-mode-map`.
<s>Inverting the order of definition fixes the issue.</s>
EDIT: Nope, it doesn't.

**Comment by wasamasa** [comment]
I'm not going to install pdf-tools just to repro a non-minimal example with a "Nope, it doesn't" comment.

**Comment by Ambrevar** [comment]
Here is the minimal example run from `make emacs` (assuming `pdf-tools` and its dependency `tablist` are installed in `package-user-dir`):
```
(require 'package)
(setq package-user-dir "~/projects")
(setq package-load-list '((pdf-tools t) (tablist t)))
(package-initialize)
(require 'pdf-tools)
(pdf-tools-install)


(evil-set-initial-state 'pdf-view-mode 'motion)
(evil-define-key 'motion pdf-view-mode-map "j" 'pdf-view-next-line-or-next-page)
(evil-set-initial-state 'image-mode 'motion)
(evil-define-key 'motion image-mode-map "j" 'image-next-line) ; TODO: This does not work.
```
Open an image file and press `j`: you'll get an error because it calls `pdf-view-next-line-or-next-page`.

I tried reproducing with `text-mode` and `outline-mode` (which inherits from `text-mode`): the issue does not occur.

**Comment by Ambrevar** [comment]
Fixed thanks to #982.
